### ISSUE TITLE
Add a 'url' attribute to MediaGalleryEntry, or use ProductImage

### ISSUE THREAD

**Issue by cjelger** [magento/graphql-ce#782]
### Description (*)

With `ProductImage.url`, one can get the URL of the main images of a product. However, `MediaGalleryEntry` only has a `file` attribute, so it is not possible to get the absolute URL of gallery images without, for example, getting the hostname from `storeConfig.base_media_url`.

While this is annoying, and if I understand correctly how the Magento GraphQL caching is done, querying `products` and `storeConfig` at the same time will disable caching because `storeConfig` is not cacheable (according to https://devdocs.magento.com/guides/v2.3/graphql/caching.html).

Which means a query like
```
{
  products(search: "yoga")
  {
    items {
      sku
      image {
        url
      }
      media_gallery_entries {
        file
      }
    }
  }
  storeConfig {
    base_media_url
  }
}
```
will not be cached because of the `storeConfig` part.

I think for consistency and also to allow caching, it would be good to either add a `url` field to `MediaGalleryEntry` or add an `image` attribute with type `ProductImage`.

### Expected behavior (*)
I could simply write (with or without the `image` attribute): 
``` 
{
  products(search: "yoga")
  {
    items {
      sku
      image {
        url
      }
      media_gallery_entries {
        image {
          url
        }
      }
    }
  }
}
```
### Benefits
This would remove the need to query `storeConfig` (and hence allow caching) and would be more consistent with the main images of a product.

### Additional information
cc: @paliarush as mentioned on Slack, and @cpartica as this would be nice for CIF

**Comment by cjelger** [comment]
I just realised that there is a similar issue if someone queries `categories` and wants to get `CategoryTree.image`. This is also just a `String` and requires the caller to also query `storeConfig`.

Of course one can query `storeConfig` once and store the result, but this doesn't work in stateless environments.

**Comment by damienwebdev** [comment]
To add to this, since you can do `video` as well as `image`...

```
{
  products(search: "yoga")
  {
    items {
      sku
      image {
        url
      }
      media_gallery_entries {
        ... on ProductImage {
            image {
                  url
            }
         }
      }
    }
  }
}
```

qualifying the query with a graphql type filter would be nice.

**Comment by prabhuram93** [comment]
@cjelger This issue is being actively worked on by the core team and is scoped for 2.3.3

**Comment by lenaorobei** [comment]
@TomashKhamlai could you please verify this issue.

**Comment by lenaorobei** [comment]
Closing since it is fixed in 2.3.3.
